test/python/transpiler/test_count_operations_pass.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""ResourceEstimation pass testing"""
+
+import unittest
+
+from qiskit import QuantumCircuit, QuantumRegister
+from qiskit.converters import circuit_to_dag
+from qiskit.transpiler.passes import ResourceEstimation
+from qiskit.test import QiskitTestCase
+
+
+class TestResourceEstimationPass(QiskitTestCase):
+    """ Tests for PropertySet methods. """
+
+    def test_empty_dag(self):
+        """ Empty DAG has 0 amount of operations """
+        circuit = QuantumCircuit()
+        dag = circuit_to_dag(circuit)
+
+        pass_ = ResourceEstimation()
+        _ = pass_.run(dag)
+
+        self.assertEqual(pass_.property_set['size'], 0)
+        self.assertEqual(pass_.property_set['depth'], 0)
+        self.assertEqual(pass_.property_set['width'], 0)
+
+    def test_count_h_and_cx(self):
```

Comment:
this pass is not actually counting h and cx, it's just checking depth and width.
I think we need a test to do that when you add `count_ops`.